### PR TITLE
Changes for building using VS2015

### LINKED ISSUE #824: Fix warnings reported by VS 2015
Fix warnings reported by VS 2015 - blocks #30 

In total there are currently 2085 warnings but some of them are reported multiple times on the same location so it is not that bad :)

We can also disable them but that might not be wise (at least not for all of them :))

Could not find a msdn link for this one
Warning	C4463	overflow; assigning 1 to bit-field that can only hold values from -1 to 0

[metamodelrw.cpp#L2755](https://github.com/dotnet/coreclr/blob/cbf46fb0b6a0b209ed1caf4a680910b383e68cba/src/md/enc/metamodelrw.cpp#L2755)
[metamodelrw.cpp#L3173](https://github.com/dotnet/coreclr/blob/cbf46fb0b6a0b209ed1caf4a680910b383e68cba/src/md/enc/metamodelrw.cpp#L3173)
[metamodelrw.cpp#L3359](https://github.com/dotnet/coreclr/blob/cbf46fb0b6a0b209ed1caf4a680910b383e68cba/src/md/enc/metamodelrw.cpp#L3359)

[Warning C4312](https://msdn.microsoft.com/en-us/library/h97f4b9y.aspx)
This warning detects an attempt to assign a 32-bit value to a 64-bit pointer type, for example, casting a 32-bit int or long to a 64-bit pointer

To much of this one to list but some of them:
'`<function-style-cast>`': conversion from 'const int' to 'IMDInternalImport *' of greater size	cee_dac
[holder.h#L659](https://github.com/dotnet/coreclr/blob/465721555eda899a06bc91fe37bc47c0a1adbdd4/src/inc/holder.h#L659)

And similar

[Warning C4311](https://msdn.microsoft.com/en-us/library/4t91x2k5.aspx)
This warning detects 64-bit pointer truncation issues. For example, if code is compiled for a 64-bit architecture, the value of a pointer (64 bits) will be truncated if it is assigned to an int (32 bits). For more information
[simplerhash.h#L450](https://github.com/dotnet/coreclr/blob/465721555eda899a06bc91fe37bc47c0a1adbdd4/src/inc/simplerhash.h#L450)

[Warning C4302](https://msdn.microsoft.com/en-us/library/c9ctb1z9.aspx)
The compiler detected a conversion from a larger type to a smaller type. Information may be lost
[simplerhash.h#L450](https://github.com/dotnet/coreclr/blob/465721555eda899a06bc91fe37bc47c0a1adbdd4/src/inc/simplerhash.h#L450)

### PR DESCRIPTION
**Pull request by jkotas**
Assorted set of changes to make build produce coreclr.dll using VS2015 successfully. Fixes #824.

More fixes are needed in the managed part of the build to make build work end to end on VS2015.

### CODE REVIEW

File: src/gc/gc.cpp [hunk 1]
```diff
@@ -19820,7 +19820,9 @@ void gc_heap::realloc_plan_generation_start (generation* gen, generation* consin
     }
 
     dprintf (1, ("plan re-alloc gen%d start at %Ix (ptr: %Ix, limit: %Ix)", gen->gen_num, 
-        generation_allocation_pointer (consing_gen), generation_allocation_limit (consing_gen))); 
+		generation_plan_allocation_start (consing_gen),
+        generation_allocation_pointer (consing_gen), 
+		generation_allocation_limit (consing_gen))); 
 }
```
Comment (DickvdBrink) [inline src/gc/gc.cpp]: Looks like mixed tab/spaces

Comment (jkotas) [inline src/gc/gc.cpp]: Fixed.

**Comment by jkotas** [comment]
fmodf - fixed. I have also cleanup how fabsf is done in the PAL while I was on it.

**Comment by jkotas** [comment]
@janvorli Could you please take a look?

**Comment by kangaroo** [comment]
@jkotas fmodf changes broke the Linux build. Once that's resolve this LGTM

**Comment by janvorli** [comment]
LGTM

**Pull request merged**